### PR TITLE
fixes #1039 - Don't run periodic tasks on non-leaders

### LINKED ISSUE #1039: Task expunge mechanism writes into ZooKeeper as non-leader
The task expunge mechanism writes to the ZooKeeper database even when it's not the leader. It expunges the tasks (all of them!) and the leader falls over it when the next TASK_RUNNING messages arrive. Because those tasks cannot be found in the task store, the leader will kill them.

=> kaboom. You have an empty cluster.

On the non-leader:
```
Jan 19 14:17:54 srv003 marathon[12628]: [2015-01-19 14:17:54,278] INFO Not leader therefore not reconciling tasks (mesosphere.marathon.MarathonSchedulerService:311)
Jan 19 14:18:24 srv003 marathon[12628]: [2015-01-19 14:18:24,285] INFO Not leader therefore not reconciling tasks (mesosphere.marathon.MarathonSchedulerService:311)
Jan 19 14:18:24 srv003 marathon[12628]: [2015-01-19 14:18:24,287] INFO Expunging orphaned tasks from store (mesosphere.marathon.tasks.TaskTracker:170)
Jan 19 14:18:24 srv003 marathon[12628]: [2015-01-19 14:18:24,320] INFO Expunging orphaned task with key task:app-master_databases_redis:app-master_databases_redis.aa0b622d-9fe4-11e4-8a66-4adee68efb3d (mesosphere.marathon.tasks.TaskTracker:178)
Jan 19 14:18:24 srv003 marathon[12628]: [2015-01-19 14:18:24,341] INFO Expunging orphaned task with key task:app-master_services_beat:app-master_services_beat.c033fdc0-9fe4-11e4-8a66-4adee68efb3d (mesosphere.marathon.tasks.TaskTracker:178)
Jan 19 14:18:24 srv003 marathon[12628]: [2015-01-19 14:18:24,383] INFO Expunging orphaned task with key task:app-master_services_flower:app-master_services_flower.90e11f57-9fe4-11e4-8a66-4adee68efb3d (mesosphere.marathon.tasks.TaskTracker:178)
```

On the leader:
```
Jan 19 14:18:51 srv002 marathon[19022]: [2015-01-19 14:18:51,993] INFO Received status update for task app-staging_services_beat.02e73ab4-9fe6-11e4-8f20-b6a514b75aa4: TASK_RUNNING (Reconciliation: Latest task state) (mesosphere.marathon.MarathonScheduler:148)
Jan 19 14:18:52 srv002 marathon[19022]: [2015-01-19 14:18:52,013] WARN Task for ID app-staging_services_beat.02e73ab4-9fe6-11e4-8f20-b6a514b75aa4 already running, ignoring (mesosphere.marathon.tasks.TaskTracker:85)
Jan 19 14:18:52 srv002 marathon[19022]: [2015-01-19 14:18:52,014] WARN Couldn't post event for value: "app-staging_services_beat.02e73ab4-9fe6-11e4-8f20-b6a514b75aa4"
Jan 19 14:18:52 srv002 marathon[19022]:  (mesosphere.marathon.MarathonScheduler:188)
Jan 19 14:18:52 srv002 marathon[19022]: java.lang.Exception: Task for ID app-staging_services_beat.02e73ab4-9fe6-11e4-8f20-b6a514b75aa4 already running, ignoring
Jan 19 14:18:52 srv002 marathon[19022]: #011at mesosphere.marathon.tasks.TaskTracker.running(TaskTracker.scala:86)
Jan 19 14:18:52 srv002 marathon[19022]: #011at mesosphere.marathon.MarathonScheduler.statusUpdate(MarathonScheduler.scala:180)
Jan 19 14:18:52 srv002 marathon[19022]: [2015-01-19 14:18:52,014] WARN Killing task value: "app-staging_services_beat.02e73ab4-9fe6-11e4-8f20-b6a514b75aa4"
Jan 19 14:18:52 srv002 marathon[19022]:  (mesosphere.marathon.MarathonScheduler:189)
```

### PR DESCRIPTION
**Pull request by drexin**

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/MarathonSchedulerService.scala [hunk 1]
```diff
@@ -258,6 +257,8 @@ class MarathonSchedulerService @Inject() (
 
     schedulerActor ! Suspend(LostLeadershipException("Leadership was defeated"))
 
+    reconciliationTimer.cancel()
```
Comment (ConnorDoyle) [inline src/main/scala/mesosphere/marathon/MarathonSchedulerService.scala]: This invalidates the timer, causing any subsequent calls to `schedule` to throw an `IllegalStateException`.  We need to create a new timer at some point.  See: http://docs.oracle.com/javase/7/docs/api/java/util/Timer.html#cancel()

Comment (drexin) [inline src/main/scala/mesosphere/marathon/MarathonSchedulerService.scala]: Good catch, thanks!

**Comment by ConnorDoyle** [comment]
Could you add a test to verify that (`onElected`, `onDefeated`, `onElected`) works?  Can also smoke-test it using the new `DELETE /v2/leader`.  Other than that, LGTM.

**Comment by sttts** [comment]
LGTM. Testing it in a minute in our test environment.

**Comment by ConnorDoyle** [comment]
:100:

**Pull request merged**